<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>2160</width>
    <height>1350</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <widget class="QPushButton" name="PB_AddressList">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>0</y>
      <width>1080</width>
      <height>100</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>微软雅黑 Light</family>
      <pointsize>16</pointsize>
      <weight>75</weight>
      <bold>true</bold>
     </font>
    </property>
    <property name="text">
     <string>通讯录</string>
    </property>
    <property name="iconSize">
     <size>
      <width>16</width>
      <height>16</height>
     </size>
    </property>
   </widget>
   <widget class="QPushButton" name="PB_Text">
    <property name="geometry">
     <rect>
      <x>1080</x>
      <y>0</y>
      <width>1080</width>
      <height>100</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>微软雅黑 Light</family>
      <pointsize>16</pointsize>
      <weight>75</weight>
      <bold>true</bold>
     </font>
    </property>
    <property name="text">
     <string>文本统计</string>
    </property>
   </widget>
   <widget class="QPushButton" name="PB_AllAdress">
    <property name="enabled">
     <bool>true</bool>
    </property>
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>100</y>
      <width>720</width>
      <height>90</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>微软雅黑 Light</family>
      <pointsize>12</pointsize>
      <weight>75</weight>
      <bold>true</bold>
     </font>
    </property>
    <property name="text">
     <string>全部</string>
    </property>
    <property name="iconSize">
     <size>
      <width>16</width>
      <height>16</height>
     </size>
    </property>
   </widget>
   <widget class="QPushButton" name="PB_QueryAdress">
    <property name="geometry">
     <rect>
      <x>1440</x>
      <y>100</y>
      <width>720</width>
      <height>90</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>微软雅黑 Light</family>
      <pointsize>12</pointsize>
      <weight>75</weight>
      <bold>true</bold>
     </font>
    </property>
    <property name="text">
     <string>查询</string>
    </property>
    <property name="iconSize">
     <size>
      <width>16</width>
      <height>16</height>
     </size>
    </property>
   </widget>
   <widget class="QPushButton" name="PB_GroupAdress">
    <property name="geometry">
     <rect>
      <x>720</x>
      <y>100</y>
      <width>720</width>
      <height>90</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>微软雅黑 Light</family>
      <pointsize>12</pointsize>
      <weight>75</weight>
      <bold>true</bold>
     </font>
    </property>
    <property name="text">
     <string>分组</string>
    </property>
    <property name="iconSize">
     <size>
      <width>16</width>
      <height>16</height>
     </size>
    </property>
   </widget>
   <widget class="QPushButton" name="PB_TextInfo">
    <property name="enabled">
     <bool>true</bool>
    </property>
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>100</y>
      <width>720</width>
      <height>90</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>微软雅黑 Light</family>
      <pointsize>12</pointsize>
      <weight>75</weight>
      <bold>true</bold>
     </font>
    </property>
    <property name="text">
     <string>统计</string>
    </property>
    <property name="iconSize">
     <size>
      <width>16</width>
      <height>16</height>
     </size>
    </property>
    <property name="default">
     <bool>false</bool>
    </property>
   </widget>
   <widget class="QPushButton" name="PB_DiffHashfun">
    <property name="enabled">
     <bool>true</bool>
    </property>
    <property name="geometry">
     <rect>
      <x>720</x>
      <y>100</y>
      <width>720</width>
      <height>90</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>微软雅黑 Light</family>
      <pointsize>12</pointsize>
      <weight>75</weight>
      <bold>true</bold>
     </font>
    </property>
    <property name="text">
     <string>比较不同哈希函数</string>
    </property>
    <property name="iconSize">
     <size>
      <width>16</width>
      <height>16</height>
     </size>
    </property>
   </widget>
   <widget class="QPushButton" name="PB_DiffConflict">
    <property name="enabled">
     <bool>true</bool>
    </property>
    <property name="geometry">
     <rect>
      <x>1440</x>
      <y>100</y>
      <width>720</width>
      <height>90</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>微软雅黑 Light</family>
      <pointsize>12</pointsize>
      <weight>75</weight>
      <bold>true</bold>
     </font>
    </property>
    <property name="text">
     <string>比较不同处理冲突方法</string>
    </property>
    <property name="iconSize">
     <size>
      <width>16</width>
      <height>16</height>
     </size>
    </property>
   </widget>
   <widget class="QListWidget" name="List_AddressBook">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>190</y>
      <width>1080</width>
      <height>1160</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>微软雅黑</family>
      <pointsize>12</pointsize>
     </font>
    </property>
   </widget>
   <widget class="QTextEdit" name="Text_MemberInfo">
    <property name="geometry">
     <rect>
      <x>1080</x>
      <y>190</y>
      <width>1080</width>
      <height>580</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>微软雅黑</family>
      <pointsize>18</pointsize>
     </font>
    </property>
   </widget>
   <widget class="QPushButton" name="PB_DeleteMember">
    <property name="geometry">
     <rect>
      <x>1970</x>
      <y>690</y>
      <width>181</width>
      <height>71</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>微软雅黑</family>
      <pointsize>16</pointsize>
     </font>
    </property>
    <property name="text">
     <string>删除</string>
    </property>
   </widget>
   <widget class="QPushButton" name="PB_AddMember">
    <property name="geometry">
     <rect>
      <x>1240</x>
      <y>690</y>
      <width>181</width>
      <height>71</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>微软雅黑</family>
      <pointsize>16</pointsize>
     </font>
    </property>
    <property name="text">
     <string>添加</string>
    </property>
   </widget>
   <widget class="QTextEdit" name="Text_OverallInfo">
    <property name="geometry">
     <rect>
      <x>1080</x>
      <y>770</y>
      <width>1080</width>
      <height>580</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>微软雅黑</family>
      <pointsize>18</pointsize>
     </font>
    </property>
   </widget>
   <widget class="QPushButton" name="PB_ChangeMember">
    <property name="geometry">
     <rect>
      <x>1780</x>
      <y>690</y>
      <width>181</width>
      <height>71</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>微软雅黑</family>
      <pointsize>16</pointsize>
     </font>
    </property>
    <property name="text">
     <string>更改</string>
    </property>
   </widget>
   <widget class="QPushButton" name="PB_ManageGroup">
    <property name="geometry">
     <rect>
      <x>1970</x>
      <y>690</y>
      <width>181</width>
      <height>71</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>微软雅黑</family>
      <pointsize>16</pointsize>
     </font>
    </property>
    <property name="text">
     <string>管理分组</string>
    </property>
   </widget>
   <widget class="QPushButton" name="PB_CreateGroup">
    <property name="geometry">
     <rect>
      <x>1970</x>
      <y>780</y>
      <width>181</width>
      <height>71</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>微软雅黑</family>
      <pointsize>16</pointsize>
     </font>
    </property>
    <property name="text">
     <string>新建分组</string>
    </property>
   </widget>
   <widget class="QPushButton" name="PB_QueryMember">
    <property name="geometry">
     <rect>
      <x>1970</x>
      <y>690</y>
      <width>181</width>
      <height>71</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>微软雅黑</family>
      <pointsize>16</pointsize>
     </font>
    </property>
    <property name="text">
     <string>查询</string>
    </property>
   </widget>
   <widget class="QPushButton" name="PB_SortName">
    <property name="geometry">
     <rect>
      <x>1090</x>
      <y>780</y>
      <width>280</width>
      <height>120</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>微软雅黑</family>
      <pointsize>16</pointsize>
     </font>
    </property>
    <property name="text">
     <string>按姓名排序</string>
    </property>
   </widget>
   <widget class="QPushButton" name="PB_SortPhone">
    <property name="geometry">
     <rect>
      <x>1090</x>
      <y>910</y>
      <width>280</width>
      <height>120</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>微软雅黑</family>
      <pointsize>16</pointsize>
     </font>
    </property>
    <property name="text">
     <string>按手机号排序</string>
    </property>
   </widget>
   <widget class="QPushButton" name="PB_SortUp">
    <property name="geometry">
     <rect>
      <x>1090</x>
      <y>1040</y>
      <width>280</width>
      <height>120</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>微软雅黑</family>
      <pointsize>16</pointsize>
     </font>
    </property>
    <property name="text">
     <string>升序</string>
    </property>
   </widget>
   <widget class="QPushButton" name="PB_SortDown">
    <property name="geometry">
     <rect>
      <x>1090</x>
      <y>1170</y>
      <width>280</width>
      <height>120</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>微软雅黑</family>
      <pointsize>16</pointsize>
     </font>
    </property>
    <property name="text">
     <string>降序</string>
    </property>
   </widget>
   <widget class="QPushButton" name="PB_OpenAdressBook">
    <property name="geometry">
     <rect>
      <x>1780</x>
      <y>780</y>
      <width>181</width>
      <height>71</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>微软雅黑</family>
      <pointsize>16</pointsize>
     </font>
    </property>
    <property name="text">
     <string>打开</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="Input_QueryMember">
    <property name="geometry">
     <rect>
      <x>1090</x>
      <y>690</y>
      <width>871</width>
      <height>71</height>
     </rect>
    </property>
   </widget>
   <widget class="QPushButton" name="PB_QueryInfo">
    <property name="geometry">
     <rect>
      <x>1870</x>
      <y>780</y>
      <width>280</width>
      <height>120</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>微软雅黑</family>
      <pointsize>16</pointsize>
     </font>
    </property>
    <property name="text">
     <string>统计查询效率</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="Input_CreateGroup">
    <property name="geometry">
     <rect>
      <x>1090</x>
      <y>780</y>
      <width>871</width>
      <height>71</height>
     </rect>
    </property>
   </widget>
   <widget class="QListWidget" name="List_MemberInGroup">
    <property name="geometry">
     <rect>
      <x>1080</x>
      <y>859</y>
      <width>1080</width>
      <height>491</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>微软雅黑</family>
      <pointsize>12</pointsize>
     </font>
    </property>
   </widget>
   <widget class="QTextEdit" name="Text_ArticleShow">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>190</y>
      <width>1080</width>
      <height>1160</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>微软雅黑</family>
      <pointsize>12</pointsize>
     </font>
    </property>
   </widget>
   <widget class="QTextEdit" name="Text_ArticalInfo">
    <property name="geometry">
     <rect>
      <x>1080</x>
      <y>190</y>
      <width>1080</width>
      <height>580</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>微软雅黑</family>
      <pointsize>18</pointsize>
     </font>
    </property>
   </widget>
   <widget class="QListWidget" name="List_WordInfo">
    <property name="geometry">
     <rect>
      <x>1080</x>
      <y>770</y>
      <width>1080</width>
      <height>580</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>微软雅黑</family>
      <pointsize>12</pointsize>
     </font>
    </property>
   </widget>
   <widget class="QPushButton" name="PB_OpenArtical">
    <property name="geometry">
     <rect>
      <x>1970</x>
      <y>690</y>
      <width>181</width>
      <height>71</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>微软雅黑</family>
      <pointsize>16</pointsize>
     </font>
    </property>
    <property name="text">
     <string>打开</string>
    </property>
   </widget>
   <widget class="QListWidget" name="List_HashFun">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>190</y>
      <width>1080</width>
      <height>580</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>微软雅黑</family>
      <pointsize>16</pointsize>
     </font>
    </property>
   </widget>
   <widget class="QTextEdit" name="Text_HashInfo">
    <property name="geometry">
     <rect>
      <x>1080</x>
      <y>190</y>
      <width>1080</width>
      <height>580</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>微软雅黑</family>
      <pointsize>18</pointsize>
     </font>
    </property>
   </widget>
   <widget class="QPushButton" name="PB_HashLinear">
    <property name="geometry">
     <rect>
      <x>1090</x>
      <y>780</y>
      <width>450</width>
      <height>120</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>微软雅黑</family>
      <pointsize>16</pointsize>
     </font>
    </property>
    <property name="text">
     <string>开放定值法-线性探测</string>
    </property>
   </widget>
   <widget class="QPushButton" name="PB_HashQsr">
    <property name="geometry">
     <rect>
      <x>1090</x>
      <y>910</y>
      <width>450</width>
      <height>120</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>微软雅黑</family>
      <pointsize>16</pointsize>
     </font>
    </property>
    <property name="text">
     <string>开放定值法-平方探测</string>
    </property>
   </widget>
   <widget class="QPushButton" name="PB_HashRandom">
    <property name="geometry">
     <rect>
      <x>1090</x>
      <y>1040</y>
      <width>450</width>
      <height>120</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>微软雅黑</family>
      <pointsize>16</pointsize>
     </font>
    </property>
    <property name="text">
     <string>开放定值法-随机探测</string>
    </property>
   </widget>
   <widget class="QPushButton" name="PB_HashLink">
    <property name="geometry">
     <rect>
      <x>1090</x>
      <y>1170</y>
      <width>450</width>
      <height>120</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>微软雅黑</family>
      <pointsize>16</pointsize>
     </font>
    </property>
    <property name="text">
     <string>链地址法</string>
    </property>
   </widget>
   <widget class="QPushButton" name="PB_HashNewRule">
    <property name="geometry">
     <rect>
      <x>1680</x>
      <y>780</y>
      <width>470</width>
      <height>120</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>微软雅黑</family>
      <pointsize>16</pointsize>
     </font>
    </property>
    <property name="text">
     <string>设置哈希表规则</string>
    </property>
   </widget>
   <widget class="QListWidget" name="List_HashFunCompare">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>770</y>
      <width>1080</width>
      <height>580</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>微软雅黑</family>
      <pointsize>16</pointsize>
     </font>
    </property>
   </widget>
   <widget class="QPushButton" name="PB_DiffSqr">
    <property name="geometry">
     <rect>
      <x>1090</x>
      <y>780</y>
      <width>450</width>
      <height>120</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>微软雅黑</family>
      <pointsize>16</pointsize>
     </font>
    </property>
    <property name="text">
     <string>平方取中法</string>
    </property>
   </widget>
   <widget class="QPushButton" name="PB_DiffMod">
    <property name="geometry">
     <rect>
      <x>1090</x>
      <y>910</y>
      <width>450</width>
      <height>120</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>微软雅黑</family>
      <pointsize>16</pointsize>
     </font>
    </property>
    <property name="text">
     <string>去留余数法</string>
    </property>
   </widget>
   <widget class="QPushButton" name="PB_DiffRandom">
    <property name="geometry">
     <rect>
      <x>1090</x>
      <y>1040</y>
      <width>450</width>
      <height>120</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>微软雅黑</family>
      <pointsize>16</pointsize>
     </font>
    </property>
    <property name="text">
     <string>随机数法</string>
    </property>
   </widget>
   <widget class="QListWidget" name="List_HashDiff">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>190</y>
      <width>1080</width>
      <height>580</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>微软雅黑</family>
      <pointsize>16</pointsize>
     </font>
    </property>
   </widget>
   <widget class="QListWidget" name="List_HashDiffCompare">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>770</y>
      <width>1080</width>
      <height>580</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>微软雅黑</family>
      <pointsize>16</pointsize>
     </font>
    </property>
   </widget>
   <widget class="QTextEdit" name="Text_HashDiffInfo">
    <property name="geometry">
     <rect>
      <x>1080</x>
      <y>190</y>
      <width>1080</width>
      <height>580</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>微软雅黑</family>
      <pointsize>18</pointsize>
     </font>
    </property>
   </widget>
   <widget class="QLineEdit" name="Input_setHashSIZE">
    <property name="geometry">
     <rect>
      <x>1680</x>
      <y>930</y>
      <width>471</width>
      <height>91</height>
     </rect>
    </property>
   </widget>
   <zorder>Input_CreateGroup</zorder>
   <zorder>List_MemberInGroup</zorder>
   <zorder>List_WordInfo</zorder>
   <zorder>Text_HashInfo</zorder>
   <zorder>Text_ArticalInfo</zorder>
   <zorder>PB_AddressList</zorder>
   <zorder>PB_Text</zorder>
   <zorder>PB_AllAdress</zorder>
   <zorder>PB_QueryAdress</zorder>
   <zorder>PB_GroupAdress</zorder>
   <zorder>PB_TextInfo</zorder>
   <zorder>PB_DiffHashfun</zorder>
   <zorder>PB_DiffConflict</zorder>
   <zorder>List_AddressBook</zorder>
   <zorder>Text_MemberInfo</zorder>
   <zorder>PB_DeleteMember</zorder>
   <zorder>Text_OverallInfo</zorder>
   <zorder>PB_ManageGroup</zorder>
   <zorder>PB_CreateGroup</zorder>
   <zorder>PB_QueryMember</zorder>
   <zorder>PB_SortName</zorder>
   <zorder>PB_SortPhone</zorder>
   <zorder>PB_SortUp</zorder>
   <zorder>PB_SortDown</zorder>
   <zorder>PB_OpenAdressBook</zorder>
   <zorder>Input_QueryMember</zorder>
   <zorder>PB_QueryInfo</zorder>
   <zorder>Text_ArticleShow</zorder>
   <zorder>PB_OpenArtical</zorder>
   <zorder>List_HashFun</zorder>
   <zorder>PB_HashLinear</zorder>
   <zorder>PB_HashQsr</zorder>
   <zorder>PB_HashRandom</zorder>
   <zorder>PB_HashLink</zorder>
   <zorder>PB_HashNewRule</zorder>
   <zorder>List_HashFunCompare</zorder>
   <zorder>PB_DiffSqr</zorder>
   <zorder>PB_DiffMod</zorder>
   <zorder>PB_DiffRandom</zorder>
   <zorder>List_HashDiff</zorder>
   <zorder>List_HashDiffCompare</zorder>
   <zorder>Text_HashDiffInfo</zorder>
   <zorder>Input_setHashSIZE</zorder>
   <zorder>PB_AddMember</zorder>
   <zorder>PB_ChangeMember</zorder>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections/>
</ui>
